app/Platform/Services/TicketListService.php

```diff
@@ -0,0 +1,214 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Services;
+
+use App\Community\Enums\TicketType;
+use App\Enums\Permissions;
+use App\Models\Ticket;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Http\Request;
+
+class TicketListService
+{
+    public int $totalTickets = 0;
+    public int $numFilteredTickets = 0;
+    public int $perPage = 0;
+    public int $pageNumber = 0;
+    public int $totalPages = 0;
+
+    public function getFilterOptions(Request $request): array
+    {
+        if ($this->perPage !== 0) {
+            $validatedData = $request->validate([
+                'page.number' => 'sometimes|integer|min:1',
+            ]);
+            $this->pageNumber = (int) ($validatedData['page']['number'] ?? 1);
+        }
+
+        $validatedData = $request->validate([
+            'filter.status' => 'sometimes|string|in:all,unresolved,resolved',
+            'filter.type' => 'sometimes|integer|min:0|max:2',
+            'filter.achievement' => 'sometimes|string|in:all,core,unofficial',
+            'filter.mode' => 'sometimes|string|in:all,hardcore,softcore,unspecified',
+            'filter.developerType' => 'sometimes|string|in:all,active,junior,inactive',
+        ]);
+
+        return [
+            'status' => $validatedData['filter']['status'] ?? 'unresolved',
+            'type' => (int) ($validatedData['filter']['type'] ?? 0),
+            'achievement' => $validatedData['filter']['achievement'] ?? 'all',
+            'mode' => $validatedData['filter']['mode'] ?? 'all',
+            'developerType' => $validatedData['filter']['developerType'] ?? 'all',
+        ];
+    }
+
+    public function getSelectFilters(bool $showStatus = true, bool $showDevType = true, bool $showAchievementType = true): array
```

Comment:
Not entirely sure where it's all coming from, but the first select filter specifically has too much left padding.

**Ticket manager**
![Screenshot 2024-05-06 at 4 56 46 PM](https://github.com/RetroAchievements/RAWeb/assets/3984985/0e0dd823-db0f-4736-bcac-66e18a194229)

**Game lists**
![Screenshot 2024-05-06 at 4 57 28 PM](https://github.com/RetroAchievements/RAWeb/assets/3984985/19447533-7f11-4271-a410-3e8ea12a84e9)

This may be resolvable by modifying the classnames for a specific div in _components/meta-panel/index.blade.php_:

```html
{{-- i added the last classname --}}
<div class="grid sm:flex gap-y-4 sm:divide-x-2 divide-embed-highlight [&>*:first-child]:!pl-0">
```

This classname forces the first child element to always have a left padding value of 0.

Reply:
It's a styling difference between Sorts and Filters:
https://github.com/RetroAchievements/RAWeb/blob/1689d8cd943551457a0c45ca5eed15f440e98487/resources/views/components/meta-panel/index.blade.php#L32-L33
https://github.com/RetroAchievements/RAWeb/blob/1689d8cd943551457a0c45ca5eed15f440e98487/resources/views/components/meta-panel/index.blade.php#L41-L43

Reply:
Thanks for sharing. That is indeed the problem.

I believe I have an elegant solution to this problem.

The solution is to remove all the padding classes on those child divs so they all read as:
```html
<div class="grid gap-y-1">
```

and change the parent div to:
```html
<div
    class="
        grid sm:flex gap-y-4 sm:divide-x-2 divide-embed-highlight
        sm:*:px-4 xl:*:px-8 first:*:pl-0 last:*:pr-0
    "
>
```

_However_, this `*:` syntax is only supported with Tailwind v3.4, which is a few minor releases above our version. There are no breaking changes from our version to v3.4.

To upgrade, open _package.json_, and adjust line 49 to:
```
"tailwindcss": "^3.4.3",
```
Then, run `npm install` in your terminal while in the RAWeb directory. Then restart `npm run dev`.

Reply:
done